crates/solidity/outputs/cargo/crate/src/backend/types.rs

```diff
@@ -157,19 +341,20 @@ impl TypeRegistry {
                 definition_id: *definition_id,
                 location: DataLocation::Inherited,
             },
-            Type::Function {
+            Type::Function(FunctionType {
                 parameter_types,
                 return_type,
                 external,
                 kind,
                 ..
-            } => Type::Function {
+            }) => Type::Function(FunctionType {
                 definition_id: None,
+                implicit_receiver_type: None,
                 parameter_types: parameter_types.clone(),
                 return_type: *return_type,
                 external: *external,
                 kind: *kind,
-            },
+            }),
 
             Type::Address { .. }
```

Comment:
I'm not sure I understand how rational types are handled now. For example, how is a binary expression that results in a rational number resolved?

Reply:
Binary integer expressions are implicitly converted to the "biggest" type of either operand. For rationals/literals, the order is `Zero < HexInteger < DecimalInteger < Rational`.